### PR TITLE
[expo-updates][android] Refactor JS event queueing and emitting

### PR DESCRIPTION
**Pull request by wschurman**
# Why

The idea behind this PR is that a) the controllers shouldn't be responsible for managing their event queue, and b) that event queue shouldn't be global (it should be owned by the controller).

Secondarily, there shouldn't be a need to send the queued events in `onStartupProcedureFinished` since they are sent when the first subscriber subscribes via `OnStartObserving` method in the Module.

# How

- Create a new event manager concept that manages the queue of events.
- Now that the only events emitted are state change events, refactor it a bit to integrate more closely with the state machine and have better types.
- Make it clearer that the dev launcher controller is no-op for events.

# Test Plan

E2E tests test this functionality via the state machine events.

# Checklist

- [ ] Documentation is up to date to reflect these changes (eg: https://docs.expo.dev and README.md).
- [ ] Conforms with the [Documentation Writing Style Guide](https://github.com/expo/expo/blob/main/guides/Expo%20Documentation%20Writing%20Style%20Guide.md)
- [ ] This diff will work correctly for `npx expo prebuild` & EAS Build (eg: updated a module plugin).

### CODE REVIEW

**Comment by wschurman** [comment]
> [!WARNING]
> <b>This pull request is not mergeable via GitHub because a downstack PR is open. Once all requirements are satisfied, merge this PR as a stack <a href="https://app.graphite.dev/github/pr/expo/expo/31818?utm_source=stack-comment-downstack-mergeability-warning" >on Graphite</a>.</b>
> <a href="https://graphite.dev/docs/merge-pull-requests">Learn more</a>

* **#31818** <a href="https://app.graphite.dev/github/pr/expo/expo/31818?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* **#31805** <a href="https://app.graphite.dev/github/pr/expo/expo/31805?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* `main`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @wschurman and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: packages/expo-updates/android/src/main/java/expo/modules/updates/EnabledUpdatesController.kt [hunk 1]
```diff
@@ -70,18 +69,11 @@ class EnabledUpdatesController(
...
-      field = value
-      UpdatesUtils.sendQueuedEventsToAppContext(value, appContext, logger)
-    }
-
   @Synchronized
   private fun onStartupProcedureFinished() {
     isStartupFinished = true
     startupEndTimeMillis = System.currentTimeMillis()
     (this@EnabledUpdatesController as java.lang.Object).notify()
-    UpdatesUtils.sendQueuedEventsToAppContext(shouldEmitJsEvents, appContext, logger)
```
Comment (wschurman) [inline packages/expo-updates/android/src/main/java/expo/modules/updates/EnabledUpdatesController.kt:84]: @Kudo - does this sound reasonable to you? My understanding is that there's no longer a need to send the queued events until `OnStartObserving` is called (which sets shouldEmitJsEvents to true and then sends the queued events)

Comment (Kudo) [reply packages/expo-updates/android/src/main/java/expo/modules/updates/EnabledUpdatesController.kt:84]: yes, it looks correct to me. i forgot why i added it here. let's try to remove the line. thanks for catching up this.

**Review by Kudo — APPROVED** [review]
great refactoring 👍

File: packages/expo-updates/android/src/main/java/expo/modules/updates/EnabledUpdatesController.kt (hunk 1 above)
Comment (wschurman) [reply packages/expo-updates/android/src/main/java/expo/modules/updates/EnabledUpdatesController.kt:84]: I think it's a remnant from a previous world. This line used to be what did all the emitting. Then you added the observer subscribe function and now I think this is obsolete.

**Pull request merged**